sno/gpkg.py

```diff
@@ -317,25 +294,95 @@ def hex_wkb_to_gpkg_geom(hex_wkb, **kwargs):
 WKB_POINT_EMPTY_LE = b"\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\xF8\x7F\x00\x00\x00\x00\x00\x00\xF8\x7F"
 
 
+def _gpkg_envelope_size(flags):
+    envelope_typ = (flags & 0b00001110) >> 1
+    if envelope_typ == 1:
+        return 32
+    elif envelope_typ in (2, 3):
+        return 48
+    elif envelope_typ == 4:
+        return 64
+    elif envelope_typ == 0:
+        return 0
+    else:
+        raise ValueError("Invalid envelope contents indicator")
+
+
+def _wkb_endianness_and_geometry_type(buf, wkb_offset=0):
+    """
+    Given a buffer containing some WKB at the given offset,
+    returns a two-tuple:
+        * is_little_endian (bool)
+        * geometry type as an integer.
```

Comment:
OGR Geometry type enum? WKB Geometry Type identifier?

Reply:
afaict those are the same thing :) but i'll clarify this is the type extracted directly from the wkb